### PR TITLE
Adding generator op ConstantLike

### PR DESCRIPTION
**Pull request by spandantiwari**
This is a generator op for producing tensors with constant values (e.g. ones, zeros), that are useful for models. The shape and datatype of the generated tensor can be specified through an input tensor or through attributes.

Please note that there is a similar experimental op `ConstantFill`. This op is a somewhat streamlined version with more datatype support.

### CODE REVIEW

**Comment by spandantiwari** [comment]
Not sure what the failure is. @houseroad could you please point to the issue? Is it related to test coverage?

**Comment by houseroad** [comment]
Hi @spandantiwari, you probably need to run `python onnx/backend/test/stat_coverage.py` to update the test coverage page.

**Comment by spandantiwari** [comment]
Hi @houseroad - Thanks for the pointer. I tried running `stat_coverage.py`, but I see a `urllib.error.HTTPError: HTTP Error 403: Forbidden` error. Do I need some permission to download the models? I see that it is trying to access `*opset_9` models, which are not listed on onnx/models repo.

**Comment by spandantiwari** [comment]
@houseroad - still not getting stat_coverage to run. 
Also, ping for review...

**Comment by houseroad** [comment]
@spandantiwari sorry, the models haven't been updated yet. I will do it probably tomorrow. Could you use opset_8 models to generate the results?

File: onnx/defs/operator_sets.h [hunk 1]
```diff
@@ -23,6 +23,7 @@ class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 1, Clip);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 1, Concat);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 1, Constant);
 class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 1, ConstantFill);
+class ONNX_OPERATOR_SET_SCHEMA_CLASS_NAME(Onnx, 1, ConstantLike);
```
Comment (houseroad) [inline onnx/defs/operator_sets.h]: we should add it as 9, because it is just added, right?

File: onnx/defs/generator/defs.cc [hunk 2]
```diff
@@ -34,6 +34,91 @@ ONNX_OPERATOR_SET_SCHEMA(
...
+argument. If 'dtype' is not specified, then the type of input tensor is used. If input 
+tensor is also not specified, then the type defaults to 'float'.
+
+The 'dtype' argument must be one of the data types specified in the 'DataType' enum field in the
+TensorProto message and be valid as an output type.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    ConstantLike,
+    1,
```
Comment (houseroad) [inline onnx/defs/generator/defs.cc]: Same here: opset version 9?

**Comment by spandantiwari** [comment]
@houseroad - thanks for the feedback. I have updated the PR based on your comments. I was also able to use  opset_8 models to generate test coverage results so all the checks are now passing. Could you please review. Thanks.

**Comment by spandantiwari** [comment]
@houseroad - I have updated the branch to remove some conflicts with master (maybe from your latest Less, Greater check-in). All checks are passing. Could you please review and see if this can be approved.

File: onnx/backend/test/case/node/constantlike.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @staticmethod
+    def export_threes_with_shape_and_dtype():  # type: () -> None
+        shape = (3, 4)
+        node = onnx.helper.make_node(
+            'ConstantLike',
+            shape=shape,
+            inputs=[],
+            outputs=['y'],
+            value=3.0,
+            dtype=11,  # 11: DOUBLE
```
Comment (houseroad) [inline onnx/backend/test/case/node/constantlike.py]: Can you directly refer to the Enum, like `https://github.com/onnx/onnx/blob/master/onnx/backend/test/case/node/constant.py#L24`?

Comment (spandantiwari) [reply onnx/backend/test/case/node/constantlike.py]: Good point. Done.

**Comment by spandantiwari** [comment]
@houseroad Addressed the issue in your latest feedback.

File: onnx/defs/generator/defs.cc [hunk 4]
```diff
@@ -34,6 +34,91 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, tensor_proto);
         }));
 
+static const char* ConstantLike_ver1_doc = R"DOC(
```
[the remark this answers is not in the document]
Comment (spandantiwari) [reply onnx/defs/generator/defs.cc]: Fixed.

**Comment by spandantiwari** [comment]
@houseroad - Addressed your comments.

**Comment by spandantiwari** [comment]
@houseroad - Thanks.

**Pull request merged**